### PR TITLE
Inline policy names for IAM Groups

### PR DESCRIPTION
**Pull request by jb897**
Signed-off-by: Josh Bowman <joshua.bowman@sap.com>

### Description

Adding ability to filter AWS IAM Groups by whether they have inline policies, as well as being able to access the inline policies that are attached to individual groups.

### Issues Resolved

We needed to filter AWS IAM Groups which have inline policies connected to them, and we were unable to do so.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_group.rb [hunk 1]
```diff
@@ -21,12 +21,18 @@ def initialize(opts = {})
...
+      group_name = { group_name: opts[:group_name] }
+      iam_client = @aws.iam_client
 
       group       = @resp[:group]
       @group_name = group.group_name
       @group_id   = group.group_id
       @arn        = group.arn
       @users      = @resp[:users].map(&:user_name)
+      @inline_policy_names = iam_client.list_group_policies(group_name).policy_names
+    rescue Aws::IAM::Errors::NoSuchEntity
```
Comment (sathish-progress) [inline libraries/aws_iam_group.rb:33]: `catch_aws_errors` catches this exception automatically, so this is not required.

File: libraries/aws_iam_groups.rb [hunk 2]
```diff
@@ -11,13 +11,15 @@ class AwsIamGroups < AwsResourceBase
     end
   '
 
-  attr_reader :table
+  attr_reader :table, :inline_policy_names
```
Comment (sathish-progress) [inline libraries/aws_iam_groups.rb:14]: this is not required. The FilterTable registration takes care of creating the method .

File: libraries/aws_iam_groups.rb [hunk 3]
```diff
@@ -37,10 +39,13 @@ def fetch_data
...
-                           group_id:   g.group_id,
-                           arn:        g.arn,
-                           users:      @aws.iam_client.get_group({ 'group_name': g.group_name }).users.map(&:user_name) }]
+          inline_policies = @aws.iam_client.list_group_policies({ 'group_name': g.group_name }).policy_names
+          group_rows += [{ group_name:          g.group_name,
+                           group_id:            g.group_id,
+                           arn:                 g.arn,
+                           users:               @aws.iam_client.get_group({ 'group_name': g.group_name }).users.map(&:user_name),
+                           has_inline_policies: !inline_policies.empty?,
+                           inline_policy_names: @aws.iam_client.list_group_policies({ 'group_name': g.group_name }).policy_names }]
```
Comment (sathish-progress) [inline libraries/aws_iam_groups.rb:48]: isn't this same as `inline_policies` ?

```suggestion
                           inline_policy_names: inline_policies }]
```

**Review by sathish-progress — CHANGES_REQUESTED** [review]
please review suggestions..

File: libraries/aws_iam_group.rb (hunk 1 above)
Comment (jb897) [reply libraries/aws_iam_group.rb]: Tested and confirmed this line is not required. Thanks for the catch.

File: libraries/aws_iam_groups.rb (hunk 2 above)
Comment (jb897) [reply libraries/aws_iam_groups.rb]: Tested and confirmed this line is not required. Thanks for the catch.

File: libraries/aws_iam_groups.rb (hunk 3 above)
Comment (jb897) [reply libraries/aws_iam_groups.rb]: Tested and confirmed this line is not required. Thanks for the catch.

**Review by sathish-progress — APPROVED** [review]
LGTM

**Comment by jb897** [comment]
Are you waiting on anything from my end in order to merge this PR

**Comment by sathish-progress** [comment]
> Are you waiting on anything from my end in order to merge this PR
No, nothing from your end.

File: docs/resources/aws_iam_group.md [hunk 4]
```diff
@@ -31,12 +31,13 @@ See also the [AWS documentation on IAM Groups](https://docs.aws.amazon.com/IAM/l
...
-|group\_id   | The group ID. |
-|arn         | The Amazon Resource Name of the group. |
-|users       | Array of users associated with the group.  |
+|Property            | Description|
+| ---                | --- |
+|group\_name         | The group name. |
+|group\_id           | The group ID. |
+|arn                 | The Amazon Resource Name of the group. |
+|users               | Array of users associated with the group.  |
+|inline_policy_names | The arns of the standalone IAM policies which are attached to the user. |
```
Comment (IanMadd) [inline docs/resources/aws_iam_group.md:40]: Is this the policy name or the ARN?

Comment (soumyo13) [reply docs/resources/aws_iam_group.md:40]: The arn returns the Amazon Resource Name (ARN) for the specified AWS::IAM::Group resource. For example: arn:aws:iam::123456789012:group/mystack-mygroup-1DZETITOWEKVO.

Source: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iam-group.html

Comment (sathish-progress) [reply docs/resources/aws_iam_group.md:40]: @IanMadd, to summarise, yes it is just the policy name.

Comment (jb897) [reply docs/resources/aws_iam_group.md:40]: Good catch, will change

File: docs/resources/aws_iam_groups.md [hunk 5]
```diff
@@ -28,13 +28,15 @@ See also the [AWS documentation on IAM Groups](https://docs.aws.amazon.com/IAM/l
...
-|users        | Array of users associated with the group.  |
-|entries      | Provides access to the raw results of the query, which can be treated as an array of hashes. |
+|Property              | Description|
+| ---                  | --- |
+|group\_names          | The group name. |
+|group\_ids            | The group ID. |
+|arns                  | The Amazon Resource Name of the group. |
+|users                 | Array of users associated with the group.  |
+|entries               | Provides access to the raw results of the query, which can be treated as an array of hashes. |
+|has\_inline\_policies | Boolean indicating whether or not the Group has policies set directly on it. |
```
Comment (IanMadd) [inline docs/resources/aws_iam_groups.md:38]:
```suggestion
|has\_inline\_policies | Boolean indicating whether or not the group has policies applied to it. |
```

File: docs/resources/aws_iam_group.md [hunk 6]
```diff
@@ -31,12 +31,13 @@ See also the [AWS documentation on IAM Groups](https://docs.aws.amazon.com/IAM/l
...
-|group\_id   | The group ID. |
-|arn         | The Amazon Resource Name of the group. |
-|users       | Array of users associated with the group.  |
+|Property            | Description|
+| ---                | --- |
+|group\_name         | The group name. |
+|group\_id           | The group ID. |
+|arn                 | The Amazon Resource Name of the group. |
+|users               | Array of users associated with the group.  |
+|inline_policy_names | A list of inline policy names associated with the user. |
```
Comment (IanMadd) [inline docs/resources/aws_iam_group.md:40]:
```suggestion
|inline_policy_names | A list of inline policy names associated with the group. |
```

Comment (soumyo13) [reply docs/resources/aws_iam_group.md]: I have committed the suggestion. @IanMadd

**Review by IanMadd — APPROVED** [review]
LGTM

**Review by soumyo13 — APPROVED** [review]

**Pull request merged**